### PR TITLE
Restructuring data and preparing for new app model

### PR DESCRIPTION
**Pull request by DavidRNogueira**
Notice that I am implementing our new structure. Now the only person that logs in is the team itself. Each team has a list of contacts for them to easily send out SMS.

### CODE REVIEW

File: .idea/door2door.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (eolculnamo2) [inline .idea/door2door.iml:1]: Could we add .idea stuff to our .gitignore?

.idea/**/*

File: api/src/main/java/org/door2door/dao/impl/LoginDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.door2door.dao.LoginDao;
+import org.door2door.entities.TeamEntity;
+import org.hibernate.Session;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+
+public class LoginDaoImpl implements LoginDao {
+    EntityManager entityManager;
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/dao/impl/LoginDaoImpl.java]: I know I messed this up on the OrganizationService, but our variables for dependency injection. should be private final

private final EntityManager entityManager

File: api/src/main/java/org/door2door/entities/ContactEntity.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.util.UUID;
+
+@Entity
+@Table(name = "users")
+public class ContactEntity {
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/entities/ContactEntity.java:13]: I think we'll need an email and password as well for login purposes

File: api/src/main/java/org/door2door/entities/ContactEntity.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	@Id
+	@Column(name="id", unique = true, nullable = false)
+	private UUID id;
+
+	@Column(name="email")
+	private String fullName;
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/entities/ContactEntity.java:19]: Should this be email like the column or fullName like the variable name?

File: api/src/main/java/org/door2door/entities/TeamEntity.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Entity
+@Table(name = "organization")
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/entities/TeamEntity.java]: Lets change the db name to match.. no need to keep it organization at this point.

File: api/src/main/java/org/door2door/resources/TeamController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import org.door2door.dto.NewTeamDto;
+import org.door2door.services.NewTeamService;
+
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+@Path("/organization")
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/resources/TeamController.java]: /team

File: api/src/main/java/org/door2door/services/impl/LoginServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import org.door2door.entities.TeamEntity;
+import org.door2door.services.LoginService;
+
+import javax.inject.Inject;
+
+public class LoginServiceImpl implements LoginService {
+    private LoginDaoImpl loginDao;
+
+    @Inject
+    public LoginServiceImpl (LoginDaoImpl loginDao) {
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/services/impl/LoginServiceImpl.java:15]: make sure to add final to our parameters on this file

File: api/src/main/java/org/door2door/services/impl/NewTeamServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class NewTeamServiceImpl implements NewTeamService {
+	TeamDao teamDao;
```
Comment (eolculnamo2) [inline api/src/main/java/org/door2door/services/impl/NewTeamServiceImpl.java]: private final

File: api/src/main/java/org/door2door/entities/ContactEntity.java (hunk 3 above)
Comment (DavidRNogueira) [reply api/src/main/java/org/door2door/entities/ContactEntity.java:13]: We aren't going to do login for the members. The teams will be able to have a password to login to configure this team

**Pull request merged**